### ISSUE TITLE
Missing changelogs for each release since move to CalVer

### ISSUE THREAD

**Issue by walkerdb** [getsentry/sentry#21594]
## Summary

Every release since 20.6.0 has had "No documented changes." as its changelog entry. Obviously many things have changed since then. Could this be added back for future releases?

## Motivation

As a consumer, including changelogs makes it much easier to track what new features or fixes have been added in recent releases. This has a few benefits: 1) it makes it much more clear what benefits we get when we update the self-hosted version of sentry, 2) it makes the updates easier, since we have a better sense of what breaking changes might have gone in, and 3) it lets me the developer keep track of the direction Sentry is going in, which helps me prepare for future changes.

Most other sentry-hosted repos still include these (for example [sentry-javascript's changelog](https://github.com/getsentry/sentry-javascript/blob/master/CHANGELOG.md)), not sure why they were removed for the main repo.

**Comment by BYK** [comment]
>not sure why they were removed for the main repo.

I think if you look at the number of changes in this repo, the answer will become self-evident.

We want to resume these changelogs eventually but we don't have the capacity impelement a good process around this:

 - trying to go through hundreds, if not thousands of changes each month and manually documenting them is not feasible
 - asking every developer to add a changelog entry per PR is also quite cumbersome (we have this on some other repos which are much lower volume and even then it is a chore and a simple copy paste of the commit title).

So the best would probably be something like an automated commit log scrape with enforced, higher quality commit logs with some manual commentary to highlight notable changes each month, similar to what VS Code does.

**Comment by walkerdb** [comment]
yeah I think even just a quick manual overview of release highlights, skipping an automated commit list, would be useful. Commits can always be referenced just by going through the commit log.

Totally get the time vs capacity issue, looking forward to a future where this is worked out!

**Comment by azhurbilo** [comment]
Status: Accepted

**Comment by kmulka-medigo** [comment]
Any update on this? Would love to update to the latest version, but not sure if it is safe to do so or if it even has a chance of fixing the issue we're running into. Are there security fixes in any of these versions? Who knows?

**Comment by BYK** [comment]
@kmulka-medigo it is not possible to document every single change but we are exploring a higher-level approach for changelogs. In general we ensure upgrades between adjacent versions are always safe and provide at least 1 version (=1 month) grace period for breaking changes. You can read more here: https://develop.sentry.dev/self-hosted/#upgrading

**Comment by BYK** [comment]
This is on our radar and we want to do something about this. It may take a while but we'll get there for sure.

**Comment by SengitU** [comment]
Hello, it's great to see this ticket is accepted, thank you 🎊 .
We are thinking about migrating from an older version, from the days where there was a changelog. Before deciding on the migration, we'd like to learn about the changes. Is there any resource that I can check what has changed meanwhile, it doesn't need to be in-depth, I guess in-depth one is a topic for this issue.

**Comment by BYK** [comment]
Hi @SengitU,

It will depend on what version you are on but there are two blog posts detailing a major infra change with Sentry v10 and the new versioning system afterwards:

1. https://blog.sentry.io/2020/01/07/self-hosted-sentry-10-is-ready-to-serve-get-it-while-its-hot
2. https://blog.sentry.io/2020/06/22/self-hosted-sentry-switching-to-calver/

If you upgrade to the latest version of Sentry (21.3.0 will be out on Monday) you will get all the latest features available on https://sentry.io. For those you may look into [our blog archives](https://blog.sentry.io/archive/) (wanted to link you to a group of posts but turns out we don't have tags for blog posts).

Feel free to ping us if you have any questions or issues during the upgrade. Using [the forum](https://forum.sentry.io) would be preferred for questions and discussions and GitHub is for bug reports and feature requests.

Hope this helps.

**Comment by SengitU** [comment]
Thanks for quick response, much appreciated!

**Comment by walkerdb** [comment]
Amazing - thank you all for getting this back together!
